src/compile/legend.ts

```diff
@@ -71,6 +71,18 @@ export function title(fieldDef: FieldDef) {
   return fieldTitle(fieldDef);
 }
 
+export function format(model: Model, channel: Channel) {
+  const fieldDef = model.fieldDef(channel);
+
+  // If the channel is binned, we should not set the formatter because we have a range label
+  if (!fieldDef.bin) {
+    return {};
+  }
+
+  const legend = fieldDef.legend;
+  return model.format(channel, (typeof legend !== 'boolean' && legend.format) || undefined);
```

Comment:
should we introduce `model.legend()` here just like how we have `model.axis()` for consistency?